components/terms_of_service/terms_of_service.jsx

```diff
@@ -0,0 +1,259 @@
+            serverError: null,
+        };
+
+        this.formattedText = memoizeResult((text) => formatText(text));
+    }
+
+    componentDidMount() {
+        if (this.props.termsEnabled) {
+            this.getServiceTerms();
+        } else {
+            GlobalActions.redirectUserToDefaultTeam();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.customServiceTermsId !== this.props.customServiceTermsId) {
+            this.getServiceTerms();
+        }
+    }
+
+    getServiceTerms = () => {
+        this.setState({
+            customServiceTermsText: '',
+            loading: false,
+        });
+        getServiceTerms(
+            (data) => {
+                this.setState({
+                    customServiceTermsText: data.text,
+                    loading: false,
+                });
+            },
+            () => {
+                GlobalActions.emitUserLoggedOutEvent(`/login?extra=${Constants.GET_TERMS_ERROR}`);
+                this.setState({
+                    loading: false,
+                });
+            }
+        );
+    };
+
+    handleLogoutClick = (e) => {
+        e.preventDefault();
+        GlobalActions.emitUserLoggedOutEvent('/login');
+    };
+
+    handleAcceptTerms = () => {
+        this.setState({
+            loadingAgree: true,
+            serverError: null,
+        });
+        this.registerUserAction(
+            'true',
+            () => {
+                const query = new URLSearchParams(this.props.location.search);
+                const redirectTo = query.get('redirect_to');
+                const redirectAction = query.get('redirect_action');
+                if (redirectTo && redirectTo.match(/^\/([^/]|$)/)) {
+                    browserHistory.push(redirectTo);
+                }
+                if (redirectAction === Constants.DEFAULT_TEAM_REDIRECT) {
+                    GlobalActions.redirectUserToDefaultTeam();
+                }
+
+                // Default action if redirectTo and redirectAction is not provided.
+                // This occurs when user directly opens '/terms_of_service'
+                if (!redirectTo && !redirectAction) {
+                    GlobalActions.redirectUserToDefaultTeam();
+                }
+                this.setState({
+                    loadingAgree: false,
+                });
+            }
+        );
+    };
+
+    handleRejectTerms = () => {
+        this.setState({
+            loadingDisagree: true,
+            serverError: null,
+        });
+        this.registerUserAction(
+            'false',
+            () => {
+                GlobalActions.emitUserLoggedOutEvent(`/login?extra=${Constants.TERMS_REJECTED}`);
+                this.setState({
+                    loadingDisagree: false,
+                });
+            }
+        );
+    };
+
+    registerUserAction = (accepted, success) => {
+        updateServiceTermsStatus(
+            this.props.customServiceTermsId,
+            accepted,
```

Comment:
Could we send an actual boolean true or false vs. having to send the string `true` and `false` above?

Reply:
Done.